consensus/enclave/impl/src/lib.rs

```diff
@@ -401,7 +401,15 @@ impl SgxConsensusEnclave {
             }
 
             // The associated MintConfigTx should be valid.
-            self.validate_mint_config_txs(vec![mint_config_tx], current_block_index, config)?;
+            // Important note about the tombstone block: Since we are validating a
+            // MintConfigTx that has been previously added to the
+            // ledger, we shouldn't take the tombstone block into account. If we use
+            // current_block_index then a past MintConfigTx will eventually
+            // result in TombstoneBlockExceeded after enough blocks have been appended
+            // following it. The workaround here is to pretend that the current
+            // block index it is being validated against is within the tombstone range.
+            let config_block_index = mint_config_tx.prefix.tombstone_block.saturating_sub(1);
+            self.validate_mint_config_txs(vec![mint_config_tx], config_block_index, config)?;
```

Comment:
If I'm not Mistaken doesn't this result in MintTxs being able to be made infinitely into the future against a MintConfigTx? Should we put an upper bound on the tombstone block that can bypassed perhaps?

Reply:
As far as the enclave is concerned, yes, you could feed it any matching, still-valid (contains a valid signature by a set of governors signed by a valid mint trust root key) `MintConfigTx` and it will accept the `MintTx`. The more meaningful validation takes place in consensus - for each token only a single `MintConfigTx` can be active at any given point, so while a malicious node operator could feed its enclave an older `MintConfigTx`, for that to actually make it to the chain this will have to be coordinated over a quorum of nodes.
I believe putting limits on how far in the past the tombstone could be is undesirable since it is impossible to predict the pace at which the network moves, and any user of it could very easily influence it. Generating and submitting a `MintConfigTx` is a manual and sensitive process (read: it takes time) and we don't want people to be able to just send a bunch of txs, cause block numbers to go up, and suddenly minting is halted until new configuration is issued.

Reply:
What you cite about the downsides of limits is the more important of the issues. The case I'm imagining is if a MintTxConfig is abandoned, it presents a potential standing security risk and we might want a way to invalidate it. We can address that in the future though.

Reply:
Only if consensus is not to be trusted...